### ISSUE TITLE
Modal - Set privacy

### ISSUE THREAD

**Issue by thibaudcolas** [wagtail/wagtail#8478]
Based upon #8475. Re-implements the existing set privacy modal with the latest designs. [Design link](https://www.figma.com/proto/h67EsVXdWsfu38WGGxWfpi/Wagtail-Design-System?page-id=2649%3A26957&node-id=5172%3A30775&viewport=386%2C48%2C0.5&scaling=min-zoom&starting-point-node-id=5172%3A31712&hide-ui=1).

> For now we won't go with 3 options, we'll go with 4 because the implementation from the BE is bigger than expected.

Reference screenshot:

![Image](https://user-images.githubusercontent.com/877585/166582759-60c271e6-747a-4a48-8751-abf94100c1fc.png)

## Blockers

- This requires complete rework of ModalWorkflow, which is currently tied to the UI of Bootstrap modals
- This requires error and loading states, since the successful initial display of the modal and submission of its form requires network requests.

## Expected components

- Trigger (same as now)
- Privacy status in side panel (same as now)
- Dialog based upon #8475
- Privacy options list
- Per-option fields
- Radio button
- Checkbox
- (no implementation of heading + toggle for logged-in users dual options)

## Tasks

- Adapt current trigger for new dialog
- New dialog
- TBC – Adapt ModalWorkflow for new a11y-dialog library
- Warning message in header
- Reuse existing markup for options (adapt if necessary only)
- Update privacy within side panel on click of "Set page privacy"
- `window.alert` when there is a network / request error

## Acceptance criteria

- [ ] Matches design
- [ ] As an editor, when I change the page’s privacy in the modal, it updates in the side panel
- [ ] As an editor, page privacy options work the same as they do in Wagtail 3.0

## Open questions

- How do we integrate this with existing ModalWorkflow code.
- What error handling do we want when the privacy fails to save (network error / backend error)
- Radio & checkbox styles – do we really want those to be applied everywhere else in Wagtail?
- Check whether we could use native browser rendering (with custom size & `accent-color`)

## Testing / review plan

- Functionality test @stevedya
- Accessibility tests by @thibaudcolas
- Code review by @thibaudcolas
